docs/advanced/node-local-storage.md

```diff
@@ -0,0 +1,356 @@
+          operator: In
+          values:
+          - poc-storage-worker-0
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: worker-1-pv
+spec:
+  capacity:
+    storage: 10Gi
+  volumeMode: Filesystem
+  accessModes:
+  - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: local-storage
+  local:
+    path: /mnt/
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - poc-storage-worker-1
+```
+
+Th PV should be available:
+
+```sh
+$ kubectl get pv
+NAME          CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS    REASON   AGE
+worker-0-pv   10Gi       RWO            Retain           Available           local-storage            95m
+worker-1-pv   10Gi       RWO            Retain           Available           local-storage            95m
+```
+
+As no PVC was created yet, you should also see no PVC:
+
+```sh
+$ kubectl get pvc
+No resources found.
+```
+
+Create a simple StatefulSet that will use the volumes (via a PVC):
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: local-test
+spec:
+  serviceName: "local-service"
+  replicas: 2
+  selector:
+    matchLabels:
+      app: local-test
+  template:
+    metadata:
+      labels:
+        app: local-test
+    spec:
+      containers:
+      - name: test-container
+        image: k8s.gcr.io/busybox
+        command:
+        - "/bin/sh"
+        args:
+        - "-c"
+        - "sleep 100000"
+        volumeMounts:
+        - name: local-vol
+          mountPath: /usr/test-pod
+  volumeClaimTemplates:
+  - metadata:
+      name: local-vol
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      storageClassName: "local-storage"
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+The PVC should be listed and the PV bound:
+
+```sh
+$ kubectl get pods
+NAME           READY   STATUS    RESTARTS   AGE
+local-test-0   1/1     Running   0          54s
+local-test-1   1/1     Running   0          48s
+$ kubectl get pvc
+NAME                     STATUS   VOLUME        CAPACITY   ACCESS MODES   STORAGECLASS    AGE
+local-vol-local-test-0   Bound    worker-1-pv   10Gi       RWO            local-storage   7s
+local-vol-local-test-1   Bound    worker-0-pv   10Gi       RWO            local-storage   1s
+$ kubectl get pv
+NAME          CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                            STORAGECLASS    REASON   AGE
+worker-0-pv   10Gi       RWO            Retain           Bound    default/local-vol-local-test-1   local-storage            97m
+worker-1-pv   10Gi       RWO            Retain           Bound    default/local-vol-local-test-0   local-storage            97m
+```
+
+You can also connect to the pod and write to the volume (is mounted on
+`/usr/test-pod`).
+
+If you delete a pod, as it is part of this stateful set, it will be recreated and
```

Comment:
"StatefulSet", "stateful set", or "Stateful Set"? Just choose one style, I would tend to "Stateful Set" if we talk about it as concept and not as configuration part.

Reply:
Good point, changed it all to "Stateful Set". Thanks!